opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/aggregation/dsl/MetricAggregationBuilder.java

```diff
@@ -88,8 +95,23 @@ public MetricAggregationBuilder(ExpressionSerializer serializer) {
       NamedAggregator node, Object context) {
     Expression expression = node.getArguments().get(0);
     Expression condition = node.getDelegated().condition();
+    Boolean distinct = node.getDelegated().distinct();
     String name = node.getName();
 
+    if (distinct) {
+      if ("count".equals(node.getFunctionName().getFunctionName().toLowerCase(Locale.ROOT))) {
```

Comment:
np: probably `node.getFunctionName...toLowerCase(Locale.ROOT)` worth a variable because 1) long chain 2) used in the `switch` below.

Reply:
Done, thanks!